3_extracting_features/DPutils.py

```diff
@@ -0,0 +1,220 @@
+            "Metadata_Well": well,
+            "Metadata_Site": site,
+            "Plate_Map_Name": f"{plate}_{well}_{site}",
+        }
+        
+        # For loop to go through all of the channels to get metadata from
+        for index, channel in enumerate(channels):
+            # Create channel path that increases by one for every consecutive channel portion of the image name
+            # Count is set to one to only change the first instance that '_1_' occurs (only change channel not site)
+            channel_path = pathlib.Path(str(image_paths).replace('_1_', f'_{index+1}_'), count=1)
+
+            # Since the channel path will keep 'DAPI' for all images, these if statements will change the name to respective channel based on name
+            if '01_2_' in str(channel_path):
+                channel_path = pathlib.Path(str(channel_path).replace('DAPI', 'GFP'))
+            if '01_3_' in str(channel_path):
+                channel_path = pathlib.Path(str(channel_path).replace('DAPI', 'RFP'))
+            # Save paths to each image into the index.csv under the correct column (within dictionary) with the name of the channel
+            file_data[channel]= os.path.relpath(channel_path, DP_images_path)
+        
+        # Add Genotype and Genotype_Replicate (hard coded to 1 for NF1 data) with to file_data dictionary
+        file_data["Genotype"] = genotype
+        file_data["Genotype_Replicate"] = 1
+
+        # Append all of the data into an index.csv file
+        index_csv_data.append(file_data)
+
+    return pd.DataFrame(index_csv_data)
+
+def compile_training_locations(
+    index_csv_path: pathlib.Path,
+    segmentation_data_path: pathlib.Path,
+    save_path: pathlib.Path,
+    object: str,
+):
+    """Compile well-site-object.csv file with cell locations, saving to save_path/plate/well
+    Args:
+        index_csv_path (pathlib.Path): Path to index.csv file for object (nuc or cyto) DeepProfiler project
+        segmentation_data_path (pathlib.Path): Path to segmentation folder with .tsv locations files
+        save_path (pathlib.Path): Path to save location files
+        object (str): Object to find segmentation locations for, either "nuc" or "cyto"
+    """
+    # Reads in csv and iterate through the rows from the plate, well and site columns
+    index_csv = pd.read_csv(index_csv_path)
+    for index, row in index_csv.iterrows():
+        plate = row["Metadata_Plate"]
+        well = row["Metadata_Well"]
+        site = row["Metadata_Site"]
+
+        # Gets identifier string that matches identifier from segmented images tsvs
+        identifier_details = row["DNA"].split("/")[-1].split("_")[0:4]
+        identifier_well = identifier_details[0]
+        identifier_site = identifier_details[3]
+        identifier = f"{identifier_well}_{identifier_site}"
+
+        # Note: The name is hard coded to "Nuclei.csv" due to how the pretrained model works. Even though it should be Cytoplasm for the cyto_project,
```

Comment:
Did we ever figure this out? Is it worth changing the pretrained model code to avoid propagating confusion downstream? (it will be tough to remember that Nuclei actually means Cytoplasm!)

Reply:
We could not figure this out due to how complicated the .h5 file is for the checkpoint and the .h5 file could not be changed based on what I could find. It would be good to investigate training our own model in the future. 

I think I can fix this by downstream after running DP to have the individual run a function to change all of the location files within the cytoplasm project and change "Nuclei" to "Cytoplasm". 

@gwaybio Would you like for me to make this function? They would have to run DP in terminal then go back to the notebook which seems tedious, but downstream it could help having the change.

Reply:
I have made a function to rename all of the files, my only issue is that you now **can't run the Jupyter notebook with one click since you can't run that function until after running DP in terminal**. 

@gwaybio Should we make it into a separate notebook entirely?

Reply:
I think it would make sense to write a separate notebook (or better yet, a `.py` file that you specify to run in the `README.md` and explicitly run in the `.sh` file.

Reply:
@gwaybio I took executive decision to take this make a new notebook!